src/js/jobSubmissionCtrl.js

```diff
@@ -71,29 +94,42 @@ jobSubmissionModule.controller('jobSubmissionCtrl', ['$scope', '$log', '$timeout
 
   // Selectize field for selecting modules
   var $select = $('#modules').selectize({
-      plugins: ['remove_button'],
-      labelField: 'label',
-      searchField: 'label',
-      valueField: 'label',
-      maxItems: 30,
-      delimiter: ',',
-      selectOnTab: true,
+    plugins: ['remove_button'],
+    labelField: 'label',
+    searchField: 'label',
+    valueField: 'label',
+    maxItems: 30,
+    delimiter: ',',
+    selectOnTab: true,
   });
 
   var selectize = $select[0].selectize;
 
   // Add options
   getModules().then(function(modules) {
     selectize.addOption(modules);
+    if(loadedJob != null) {
+
+      // Put adding items in a $timeout to avoid $digest issues when the
+      // select element issues a 'change' event.
+      $timeout(function() {
+
+        angular.forEach(loadedJob.modules, function(module){
+          selectize.addItem(module);
+        });
+        selectize.refreshOptions(false);
+        selectize.refreshItems();
+
+      }, 0);
+
+    }
     selectize.refreshOptions(false);
     selectize.refreshItems();
```

Comment:
Don't refresh the options and items here.  They are refreshed above.  This will cause a digest error.